app/services/get_reviews_by_products.rb

```diff
@@ -0,0 +1,25 @@
+class GetReviewsByProducts < BaseService
+  include Pagy::Backend
+
+  attr_reader :products, :limit, :params
+
+  def initialize(products, params = {}, limit = 3)
+    @products = products
+    @params = params
+    @limit ||= limit
+  end
+
+  def call
+    results = []
+    products.each do |product|
+      reviews = product.reviews.sort_by(&:created_at)
```

Comment:
`sort_by(&:created_at)` is one of the easter eggs (hidden bugs). It's very slow when there are a lot of reviews in that product. This would be faster:
```suggestion
      reviews = product.reviews.order(:created_at)
```